### ISSUE TITLE
Web Interface broken on M1 MacBooks

### ISSUE THREAD

**Issue by csounder** [csound/web-ide#402]
When you open csound web ide - (on Chrome or Safari) on an M1 MacBook you do not see any projects or profiles - you only see some moving lines in each of the square - this happens on my laptops and all my Csound student's laptops at Berklee this past week.

**Comment by kunstmusik** [comment]
@hlolli looks like an issue with the search service:

```
Access to fetch at 'https://web-ide-search-api.csound.com/search/prod/random/projects/8' from origin 'https://ide.csound.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

**Comment by hlolli** [comment]
This issue has been re-occouring and short term fixes have been very short term. I am working on a long term fix for this problem and I apologise that I didn't reply here sooner, but I hope to have it stable and shipped before new year.

**Comment by hlolli** [comment]
I wasn't able to finish fixing this in time, I had unexpected case of pneumonia in my family and was at the hospital. I hope in a week we'll see the fixes pushed to prod.

**Comment by vlazzarini** [comment]
Thanks for this @hlolli. Thankfully, the IDE itself is fully working (well at least here), so it's not urgent really.

**Comment by Ian-Grant** [comment]
I have the same problem with Firefox 134.0 (64-bit) running on Ubuntu 24.0.1 https://ide.csound.com/ doesn't show any examples, but the web UI (e.g. "Getting Started 1" at https://ide.csound.com/editor/21zOuoahdiJj8DnT7UN8) does work, just the search box doesn't find any projects nor do any of the popular projects appear.

**Comment by hlolli** [comment]
I will try as hard as I can, the whole search database as seaparate service is being scraped off. There are bugs on develop branch that I need to fix before I can merge to master. Worst case I will sync with Steven and I buy a cloud server and make a short term solution.

**Comment by csounder** [comment]
Thank you.  Good luck.  Hope everything goes smoothly.  Can't wait to having this amazing system all working again.Dr. Richard BoulangerProfessorElectronic Production and DesignBerklee College of MusicOn Jan 16, 2025, at 5:46 AM, Hlöðver Sigurðsson ***@***.***> wrote:﻿
I will try as hard as I can, the whole search database as seaparate service is being scraped off. There are bugs on develop branch that I need to fix before I can merge to master. Worst case I will sync with Steven and I buy a cloud server and make a short term solution.

**Comment by hlolli** [comment]
@csounder we have kicked the stone down the road and the search and random project features should work again as before. The long term fix will kick in, in next few weeks. Please try hard refersh if the problem persists. Should work on all operating systems equally.

**Comment by csounder** [comment]
YES!!!  Thanks so much.  It is working again on the M1 MacBooks.  Just in
time for the start of our spring semester at Berklee.  Thanks so much,
Hlöõver.

*- Dr.B*


*Dr. Richard Boulanger*

Professor

Electronic Production and Design

*Berklee College of Music*

Professional Writing & Technology Division

**Comment by Ian-Grant** [comment]
> [@csounder](https://github.com/csounder) we have kicked the stone down the road and the search and random project features should work again as before. The long term fix will kick in, in next few weeks. Please try hard refersh if the problem persists. Should work on all operating systems equally.

Yes, it works great with Firefox now too. Thank you!
